terraform/01-jenkins-server/main.tf

```diff
@@ -0,0 +1,80 @@
+provider "aws" {
+  profile = "default"
+  region  = local.aws_region
+}
+
+locals {
+  aws_region          = "us-east-1"
+  internal_cidr_block = "10.10.10.0/24"
+  external_cidr_block = "0.0.0.0/0"
+}
+
+resource "aws_vpc" "project_vpc" {
+  cidr_block = local.internal_cidr_block
+}
+
+resource "aws_subnet" "project_subnet" {
+  vpc_id            = aws_vpc.project_vpc.id
+  cidr_block        = local.internal_cidr_block
```

Comment:
In common cases, when you create a subnet, you specify the IPv4 CIDR block for the subnet, which is a **subset** of the VPC CIDR block.
Please read for details
https://docs.aws.amazon.com/vpc/latest/userguide/VPC_Subnets.html#subnet-basics
https://docs.aws.amazon.com/quickstart/latest/vpc/architecture.html

Reply:
I did the network small.